public/app/core/services/NewFrontendAssetsChecker.ts

```diff
@@ -0,0 +1,85 @@
+import { Location } from 'history';
+
+import { getBackendSrv, getGrafanaLiveSrv, locationService } from '@grafana/runtime';
+
+export class NewFrontendAssetsChecker {
+  private hasUpdates = false;
+  private previous = '';
+  private interval: number;
+  private checked = Date.now();
+  private prevLocationPath = '';
+
+  public constructor(interval?: number) {
+    // Default to never check for updates if last check was 5 minutes ago
+    this.interval = interval ?? 1000 * 60 * 5;
+  }
+
+  public start() {
+    // Subscribe to live connection state changes and check for new assets when re-connected
+    const live = getGrafanaLiveSrv();
+
+    if (live) {
+      live.getConnectionState().subscribe((connected) => {
+        if (connected) {
+          this._checkForUpdates();
+        }
+      });
+    }
+
+    // Subscribe to location changes
+    locationService.getHistory().listen(this.locationUpdated.bind(this));
+    this.prevLocationPath = locationService.getLocation().pathname;
+  }
+
+  /**
+   * Detect when we change section or dashboard and check for new assets if we do
+   */
+  private locationUpdated(location: Location) {
+    const newLocationSegments = location.pathname.split('/');
+    const prevLocationSegments = this.prevLocationPath.split('/');
+
+    this.prevLocationPath = location.pathname;
+
+    // First check if section (first path segment) has changed
+    if (newLocationSegments[1] !== prevLocationSegments[1]) {
```

Comment:
This logic still has the problem of possibly throwing away unsaved work. A simple use case:

* I am in explore [0]= `/explore/...` writing a complex query I want to put in a dashboard or create a ML forecast (I prefer using explore at times)
* I use the "add" button. I can use this to move to a plugin with a UI extensions action point (i.e. machine learning forecast) or add this query to a dashboard.
* I move away from explore to the machine learning plugin [0] = `/a/...` or [0] = `/dashboard/...`
* My page reloads and I lost my work.

Reply:
@academo how? the add-to from dashboard uses local storage to not lose state (so it works in new tab). Going from dashboard to plugin should also not keep any state in the browser but in URL or local storage surely?

Reply:
I know Explore's add to dashboard at least works across page reloads, maybe plugin extension navigations are written in a way that they don't support new tab/reload. 

Guess we can't do this automatically then, and have to instead show an info box to the user to reload.